pkg/plugin/plugin.go

```diff
@@ -109,9 +114,50 @@ func loadNetConf(bytes []byte) (*netConf, error) {
 		return nil, fmt.Errorf("failed to load netconf: %v", err)
 	}
 
+	// default config file dir paths
+	confdirs := []string{"/etc/kubernetes/cni/net.d/ovs.d/ovs.conf", "/etc/cni/net.d/ovs.d/ovs.conf"}
+	// if netconf contains config path already, use that as first
+	if netconf.ConfigurationPath != "" {
+		confdirs = append([]string{netconf.ConfigurationPath}, confdirs...)
+	}
+	// loop through the path and parse the JSON config
+	flatnetConf := &netConf{}
+	for _, confpath := range confdirs {
+		if pathExists(confpath) {
+			jsonFile, err := os.Open(confpath)
+			if err != nil {
+				return nil, fmt.Errorf("open ovs config file %s error: %v", confpath, err)
+			}
+			defer jsonFile.Close()
+			jsonBytes, err := ioutil.ReadAll(jsonFile)
+			if err != nil {
+				return nil, fmt.Errorf("load ovs config file %s: error: %v", confpath, err)
+			}
+			if err := json.Unmarshal(jsonBytes, flatnetConf); err != nil {
+				return nil, fmt.Errorf("parse ovs config file %s: error: %v", confpath, err)
+			}
+			break
+		}
+	}
+	// merge both netconf and flatnetConf configurations
+	if err := mergo.Merge(netconf, flatnetConf); err != nil {
+		return nil, fmt.Errorf("merge with ovs config file: error: %v", err)
+	}
+
 	return netconf, nil
 }
 
+func pathExists(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true
+	}
+	if os.IsNotExist(err) {
+		return false
+	}
+	return true
```

Comment:
This happens if the path maybe exists, however, we are not able to `stat` it. Does it make sense to continue with the execution of CNI in this case? I would vote to either return the error or just panic right away.

Reply:
Yes, Correct! done.